admin-dev/themes/new-theme/js/pages/order/merge-shipment-manager.ts

```diff
@@ -0,0 +1,129 @@
+
+export default class MergeShipmentManager {
+  private formRoute = 'admin_orders_shipment_get_merge_form';
+
+  private shipmentId: number|null = null;
+
+  private orderId: number|null = null;
+
+  private router = new Router();
+
+  constructor() {
+    this.initMergeShipmentEventHandler();
+  }
+
+  initMergeShipmentEventHandler(): void {
+    const mainDiv = document.querySelector(OrderViewPageMap.mainDiv);
+
+    if (!mainDiv) {
+      throw new Error('Main container not found');
+    }
+
+    mainDiv.addEventListener('click', this.onMergeShipmentClick);
+  }
+
+  initSubmitMergeShipmentStateHandler(): void {
+    const submitBtnEl = document.querySelector(OrderViewPageMap.submitMergeShipment);
+    const shipmentSelectEl = document.querySelector(OrderViewPageMap.selectMergeShipment);
+    const checkboxes = document.querySelectorAll('.form-check-input');
+
+    if (!(submitBtnEl instanceof HTMLButtonElement)
+      || !(shipmentSelectEl instanceof HTMLSelectElement)
+      || checkboxes.length === 0) {
+      return;
+    }
+
+    const submitBtn = submitBtnEl;
+    const shipmentSelect = shipmentSelectEl;
+
+    function toggleSubmit() {
+      const atLeastOneChecked = Array.from(checkboxes).some((cb) => cb instanceof HTMLInputElement && cb.checked);
+
+      const shipmentSelected = shipmentSelect.value !== '';
+      submitBtn.disabled = !(atLeastOneChecked && shipmentSelected);
+    }
+
+    checkboxes.forEach((cb) => cb.addEventListener('change', toggleSubmit));
+    shipmentSelect.addEventListener('change', toggleSubmit);
+    toggleSubmit();
+  }
+
+  onMergeShipmentClick = (event: Event): void => {
+    const target = event.target as HTMLElement;
+
+    if (target && target.matches(OrderViewPageMap.showMergeShipmentModalBtn)) {
+      if (!target.dataset.orderId) {
+        throw new Error('impossible to retrieve order id');
+      }
+      this.orderId = Number(target.dataset.orderId);
+
+      if (!target.dataset.shipmentId) {
+        throw new Error('impossible to retrieve shipment id');
+      }
+      this.shipmentId = Number(target.dataset.shipmentId);
+
+      this.refreshMergeShipmentForm();
+    }
+  };
+
+  async refreshMergeShipmentForm(): Promise<void> {
+    const formContainer = document.querySelector('#mergeShipmentFormContainer') as HTMLElement | null;
+
+    if (!formContainer) {
+      throw new Error('Form container or skeleton element not found inside modal.');
+    }
+
+    formContainer.classList.add('skeleton skeleton--shipment-modal');
+    formContainer.innerHTML = '';
+
+    try {
+      const response = await fetch(this.router.generate(this.formRoute, {
+        orderId: this.orderId,
+        shipmentId: this.shipmentId,
+      }), {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error(await response.text());
+      }
+
+      formContainer!.innerHTML = await response.text();
```

Comment:
```suggestion
      formContainer.innerHTML = await response.text();
```
I think the `!` is uncessary because you check it on start of the function.